compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/firFactoryUtil.kt

```diff
@@ -241,5 +247,5 @@ internal fun FirClassSymbol<*>.findSamFunction(session: FirSession): FirFunction
       memberRequiredPhase = null,
     )
   if (scope.collectAllProperties().any { it.isAbstract }) return null
```

Comment:
This check should only take place when finding the SAM function, not when collecting all abstract functions. Might be why your test data changed.

Reply:
Doesn't seem like it, the test looked broken before to me so I think the updated version is correct. Good point to not exit early here though

Reply:
https://github.com/ZacSweers/metro/pull/517